File: 1524-string-matching-in-an-array/1524-string-matching-in-an-array.py
class Solution(object):
    def stringMatching(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        ans = set()
        for word in words:
            for par in words:
                if word != par and word in par:
                    ans.add(word)
        return list(ans)

File: 1524-string-matching-in-an-array/1524-string-matching-in-an-array.java
class Solution {
    public List<String> stringMatching(String[] words) {
        return Arrays.stream(words)
                    .filter(word -> Arrays.stream(words)
                            .anyMatch(parentWord -> !parentWord.equals(word) && parentWord.contains(word)))
                    .distinct()
                    .collect(Collectors.toList());
    }
}